### PR TITLE
Add constant, define, defined for fast call

### PR DESCRIPTION
**Pull request by andot**
Add constant, define, defined for fast call

### CODE REVIEW

File: zend/constant.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        char *name = (char *)constant_name.c_str();
+        int name_len = constant_name.length();
+        zval *val = value._val;
+        zval *val_free = nullptr;
+        zend_constant c;
+        // we need the tsrm_ls variable
+        TSRMLS_FETCH();
+        /* class constant, check if there is name and make sure class is valid & exists */
+        if (zend_memnstr(name, (char *)"::", sizeof("::") - 1, name + name_len)) {
+            zend_error(E_WARNING, "Class constants cannot be defined or redefined");
```
Comment (tugrul) [inline zend/constant.cpp]: `Php::warning` output stream going to be better

File: zend/constant.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                        ALLOC_INIT_ZVAL(val_free);
+                        if (Z_OBJ_HT_P(val)->cast_object(val, val_free, IS_STRING TSRMLS_CC) == SUCCESS) {
+                            val = val_free;
+                            break;
+                        }
+                    }
+                }
+                /* no break */
+            default:
+                zend_error(E_WARNING,"Constants may only evaluate to scalar values");
```
Comment (tugrul) [inline zend/constant.cpp]: `Php::warning` output stream going to be better

File: zend/constant.cpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+     *  @param  constant_name
+     *  @return value
+     */
+    Php::Value constant(const std::string &constant_name) {
+        Php::Value result;
+        // we need the tsrm_ls variable
+        TSRMLS_FETCH();
+
+        if (!zend_get_constant_ex(constant_name.c_str(), constant_name.length(), result._val, NULL, ZEND_FETCH_CLASS_SILENT TSRMLS_CC)) {
+            php_error_docref(NULL TSRMLS_CC, E_WARNING, "Couldn't find constant %s", constant_name.c_str());
```
Comment (tugrul) [inline zend/constant.cpp]: `Php::warning` output stream going to be better

File: zend/constant.cpp [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        zval *val_free = nullptr;
+        zend_constant c;
+        // we need the tsrm_ls variable
+        TSRMLS_FETCH();
+        /* class constant, check if there is name and make sure class is valid & exists */
+        if (zend_memnstr(name, (char *)"::", sizeof("::") - 1, name + name_len)) {
+            zend_error(E_WARNING, "Class constants cannot be defined or redefined");
+            return false;
+        }
+    repeat:
```
Comment (tugrul) [inline zend/constant.cpp]: It could be better with recursive call through overloaded functions to loop around HashIterator

**Comment by EmielBruijntjes** [comment]
This feature is now implemented in PHP-CPP, based on the suggestions from this pull reqest.

**Pull request closed without merging**